org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/EclipseResourceFileSystemAccess2.java

```diff
@@ -376,9 +377,17 @@ protected IContainer getContainer(OutputConfiguration outputConfig) {
 	}
 
 	protected boolean hasContentsChanged(IFile file, StringInputStream newContent) {
-		return hasContentsChanged(file, (InputStream) newContent);
+		try {
+            return hasContentsChangedNew(file.getContents(), newContent);
+        } catch (CoreException e) {
+            return true;
+        }
 	}
 
+	private static final int DEFAULT_BUFFER_SIZE = 8192;
```

Comment:
todo do we really need this?

Reply:
Are you asking about the byte[] buffer reuse or the default buffer sizes?

Reply:
i mean using a thread local

Reply:
SInce we cannot really clear the threadlocal for a random given thread, I'd either not keep the byte arrays around or use a bounded buffer e.g. with 3 entries for the bytes[]
I doubt that we'll be able to measure the difference between allocating a byte[] for every resource vs the threadlocal stuff